api/src/main/java/com/mtech/vmcs/controller/CoinController.java

```diff
@@ -28,10 +31,16 @@ public ResponseEntity<List<Coin>> createCoins(@RequestBody List<Coin> coins) {
     return new ResponseEntity<>(HttpStatus.CREATED);
   }
 
-  // TODO: Wait for later implementation
   @PostMapping("/checkCoin")
-  public ResponseEntity checkCoin(@RequestBody Coin coin) {
-    return null;
+  public ResponseEntity checkCoin(@RequestBody Coin coin) throws IllegalAccessException {
```

Comment:
1. ```@Autowired private ObjectMapper objectMapper; ``` add it after ```@Autowired private CoinService coinService;```

2. ```suggestion
    public ResponseEntity<ObjectNode> checkCoin(@RequestBody Coin coin) {
    ObjectNode objectNode = objectMapper.valueToTree(coin);
    objectNode.put("isValid", coinService.checkCoin(coin));
    return new ResponseEntity<>(objectNode, HttpStatus.OK);
  }
```

Reply:
Ok, I have done it.